### PR TITLE
Show schedule or charge-windows only if appropriate strategy is used

### LINKED ISSUE #70: Show plot of schedule or windows only if appropriate strategy is used
Show windows only if strategy "flex_window" is used.
Show schedule only if strategy "schedule" is used.

grafik

### PR DESCRIPTION
**Pull request by jakobgemassmer**
Fix #70 

**Changes proposed in this pull request**:
- The schedule or the charge-windows are now only shown in the first plot if the appropriate strategy is used.
- If neither flex_window or schedule is used, none is shown in the plot.

The following steps were realized, as well (required):
- [x] Correct linting with `fake8 path_to_script`
- [ ] Check if tests pass locally (`pytest ./src/tests.py`)

Also the following steps were realized (if applies):
- [ ] Write docstrings to your code
- [ ] For new functionalities: Explain in readthedocs
- [ ] Write test(s) for your new patch of code

*Please mark above checkboxes as following:*
- [ ] Open
- [x] Done

### CODE REVIEW

File: src/scenario.py [hunk 1]
```diff
@@ -607,19 +607,19 @@ def run(self, strategy_name, options):
...
-
-            for gcID, schedule in gcPowerSchedule.items():
-                if any(s is not None for s in schedule):
-                    # schedule exists
-                    ax.plot(xlabels, schedule, label="Schedule {}".format(gcID))
-
-            ax.plot(xlabels, totalLoad, label="total")
+
+            # draw schedule or charge-windows
+            if type(strat).__name__ == "FlexWindow":
```
Comment (stefansc1) [inline src/scenario.py:612]: You can use `strategy_name` instead (function argument of Scenario.run). Example:
`if strategy_name == "flex_window":`

**Review by Piranias — APPROVED** [review]

File: src/strategy.py [hunk 2]
```diff
@@ -32,9 +32,12 @@ def __init__(self, constants, start_time, **kwargs):
         self.interval = kwargs.get('interval')  # required
         self.current_time = start_time - self.interval
         # relative allowed difference between battery SoC and desired SoC when leaving
-        self.margin = 0.05
+        self.margin = 0.1
```
Comment (stefansc1) [inline src/strategy.py:35]: is this change intended?

Comment (jakobgemassmer) [reply src/strategy.py:35]: Yes, the simulations showed that it appears quite often, that the desired_soc is not fully reached. Therefore, I think a little bit more default tolerance ist useful.

Comment (stefansc1) [reply src/strategy.py:35]: This has nothing to do with this PR, so a separate commit would have been cleaner. However, as it is only a minor change, it can be included if you think it's beneficial. We still have our strategy check that the vehicles can really make their trip, even when leaving with less than their desired SoC.

**Review by stefansc1 — APPROVED** [review]
